crates/re_arrow_store/benches/arrow2.rs

```diff
@@ -0,0 +1,148 @@
+use itertools::Itertools;
+use re_log_types::{component_types::Point2D, datagen::build_some_point2d, DataCell};
+
+// ---
+
+criterion_group!(benches, estimated_size_bytes);
+criterion_main!(benches);
+
+// ---
+
+#[cfg(not(debug_assertions))]
+const NUM_ROWS: usize = 10_000;
+#[cfg(not(debug_assertions))]
+const NUM_INSTANCES: usize = 100;
+
+// `cargo test` also runs the benchmark setup code, so make sure they run quickly:
+#[cfg(debug_assertions)]
+const NUM_ROWS: usize = 1;
+#[cfg(debug_assertions)]
+const NUM_INSTANCES: usize = 1;
+
+// ---
+
+fn estimated_size_bytes(c: &mut Criterion) {
+    let mut group = c.benchmark_group(format!(
+        "arrow2/erased_clone/rows={NUM_ROWS}/instances={NUM_INSTANCES}"
+    ));
+    group.throughput(criterion::Throughput::Elements(NUM_ROWS as _));
+
+    {
+        fn generate_cells() -> Vec<DataCell> {
+            (0..NUM_ROWS)
+                .map(|_| DataCell::from_native(build_some_point2d(NUM_INSTANCES).as_slice()))
+                .collect()
+        }
+
+        {
+            let cells = generate_cells();
+            let total_instances = cells.iter().map(|cell| cell.num_instances()).sum::<u32>();
+            assert_eq!(total_instances, (NUM_ROWS * NUM_INSTANCES) as u32);
+
+            group.bench_function("cell/arc_erased", |b| {
+                b.iter(|| {
+                    let cells = cells.clone();
+                    assert_eq!(
+                        total_instances,
+                        cells.iter().map(|cell| cell.num_instances()).sum::<u32>()
+                    );
+                    cells
+                });
+            });
+        }
+
+        {
+            let cells = generate_cells().into_iter().map(Arc::new).collect_vec();
+            let total_instances = cells.iter().map(|cell| cell.num_instances()).sum::<u32>();
+            assert_eq!(total_instances, (NUM_ROWS * NUM_INSTANCES) as u32);
+
+            group.bench_function("cell/wrapped_in_arc", |b| {
+                b.iter(|| {
+                    let cells = cells.clone();
+                    assert_eq!(
+                        total_instances,
+                        cells.iter().map(|cell| cell.num_instances()).sum::<u32>()
+                    );
+                    cells
+                });
+            });
+        }
+
+        {
+            let cells = generate_cells();
+            let arrays = cells
+                .iter()
+                .map(|cell| {
+                    cell.as_arrow_ref()
+                        .as_any()
+                        .downcast_ref::<StructArray>()
+                        .unwrap()
+                })
+                .collect_vec();
+            let total_instances = arrays.iter().map(|array| array.len() as u32).sum::<u32>();
+            assert_eq!(total_instances, (NUM_ROWS * NUM_INSTANCES) as u32);
+
+            group.bench_function("array/downcast_first", |b| {
+                b.iter(|| {
+                    let arrays = arrays.clone();
```

Comment:
This test doesn't make sense I believe. This is an array of references (Rust analyzer confirms my reading). I don't think copying `Vec<&T>` is what you had in mind for comparison? That would ofc massively outpace everything else.

Reply:
Oh yeah nice catch, I forgot a `.clone()` up there